src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+                               gObjectTable[objectId].vromEnd - gObjectTable[objectId].vromStart, 0,
+                               &player->maskObjectLoadQueue, NULL);
+        player->maskObjectLoading++;
+    } else if (player->maskObjectLoading == 2) {
+        if (osRecvMesg(&player->maskObjectLoadQueue, NULL, OS_MESG_NOBLOCK) == 0) {
+            player->maskObjectLoading = 0;
+
+            if (player->currentMask == PLAYER_MASK_GREAT_FAIRY) {
+                s32 i;
+
+                for (i = 0; i < ARRAY_COUNT(D_801F58B0); i++) {
+                    func_80127B64(D_801F58B0[i], ARRAY_COUNT(D_801F58B0[i]), &player->bodyPartsPos[7]);
+                }
+            }
+        }
+    } else if ((player->currentMask != PLAYER_MASK_NONE) && (player->currentMask != (u8)player->maskId)) {
+        player->maskObjectLoading = 1;
+        player->maskId = player->currentMask;
+    } else if (player->currentMask == PLAYER_MASK_CIRCUS_LEADER) {
+        s32 i;
+
+        for (i = 0; i < ARRAY_COUNT(D_801F59C8); i++) {
+            D_801F59C8[i] += Rand_S16Offset(4, 23) + (s32)(fabsf(player->linearVelocity) * 50.0f);
+        }
+    }
+}
+
+void func_80122BA4(GlobalContext* globalCtx, struct_80122D44_arg1* arg1, s32 arg2, s32 alpha) {
+    if (arg2 == arg1->unk_00) {
+        s32 index;
+
+        arg1->unk_01--;
+        if (arg1->unk_01 < 0) {
+            arg1->unk_01 = 3;
+        }
+
+        index = arg1->unk_01;
+        arg1->unk_04[index].unk_00 = arg1->unk_00;
+        arg1->unk_04[index].unk_01 = alpha;
+        Matrix_CopyCurrentState(&arg1->unk_04[index].unk_04);
+
+        arg1->unk_00 = 0;
+    }
+}
+
+void func_80122C20(GlobalContext* globalCtx, struct_80122D44_arg1* arg1) {
+    struct_80122D44_arg1_unk_04* temp_v1 = &arg1->unk_04[0];
+    s32 phi_a1;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(arg1->unk_04); i++, temp_v1++) {
+        // Can't be `temp_v1->unk_01 != 0`
+        if (temp_v1->unk_01) {
+            phi_a1 = temp_v1->unk_00 == 3 ? 0x55 : 0x33;
+            if (phi_a1 >= temp_v1->unk_01) {
+                temp_v1->unk_01 = 0;
+            } else {
+                temp_v1->unk_01 -= phi_a1;
+            }
+        }
+    }
+}
+
+typedef struct {
+    /* 0x0 */ Color_RGB8 color;
+    /* 0x4 */ Gfx* dlist;
+} struct_801BFDD0; // size = 0x08
+
+extern struct_801BFDD0 D_801BFDD0[];
+#if 0
+struct_801BFDD0 D_801BFDD0[] = {
+    { { 180, 200, 255 }, object_link_goron_DL_00BDD8 },
+    { { 155, 0, 0 }, object_link_goron_DL_014690 },
+    { { 255, 0, 0 }, object_link_goron_DL_011AB8 },
+};
+#endif
+
+void func_80122D44(GlobalContext* globalCtx, struct_80122D44_arg1* arg1) {
+    struct_801BFDD0* temp_s3;
+    struct_80122D44_arg1_unk_04* phi_s2 = arg1->unk_04;
+    s32 phi_s6 = false;
+    s32 i;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    for (i = 0; i != ARRAY_COUNT(arg1->unk_04); i++) {
+        if ((phi_s2->unk_01 != 0) && (phi_s2->unk_01 != 0xFF)) {
```

Comment:
```suggestion
        if ((phi_s2->unk_01 != 0) && (phi_s2->unk_01 != 255)) {
```
`phi_s2->unk_01` from the code below is an `envAlpha` so dec. Could maybe call this `phi_s2->envAlpha` or just `phi_s2->alpha`? Could leave it for later docs though